lib/puppet/type/puppet_authorization_hocon_rule.rb

```diff
@@ -25,6 +25,31 @@
       unless val.is_a?(Hash)
         raise "Value must be a hash but was #{value.class}"
       end
+      validate_acl(val)
+    end
+
+    def validate_acl(val)
+      ["allow", "deny"].each do |rule|
+        if val.has_key?(rule)
```

Comment:
It seems like if it's not a hash we just pass it through; this is probably fine but should we check that it's a string if it's not a hash?